Makefile.am

```diff
@@ -79,45 +83,37 @@ include $(srcdir)/regenerate-unified-compilation.am
 
 ######## Generate C++ IR files from *.def files using ir-generator
 
-# make is very stupid regarding multiple files generated from the same source
-# this pattern ensures that parallel builds continue to work
-ir/ir-generated.h: irgenerator $(ir_DEF_FILES)
-	./irgenerator -t ir/gen-tree-macro.h -i ir/ir-generated.cpp $(ir_DEF_FILES) >$@ || ( rm $@ && false )
-	@$(srcdir)/tools/fixup-line-directives $@ >$@.tmp
-	@mv $@.tmp $@
-ir/gen-tree-macro.h: ir/ir-generated.h
-	@$(srcdir)/tools/fixup-line-directives $@ >$@.tmp
-	@mv $@.tmp $@
-ir/ir-generated.cpp: ir/ir-generated.h
-	@$(srcdir)/tools/fixup-line-directives $@ >$@.tmp
-	@mv $@.tmp $@
+# Add support for automake's verbosity options for the IR generator.
+V_IRGEN = $(__v_IRGEN_$(V))
+__v_IRGEN_ = $(__v_IRGEN_$(AM_DEFAULT_VERBOSITY))
+__v_IRGEN_0 = @echo "  IRGEN   " $@;
+__v_IRGEN_1 = 
+
+# We wrap the IR generator in `ylwrap` to reduce unnecessary full-project
+# rebuilds. `ylwrap` won't overwrite the generated headers unless they've
+# changed.
+ir/ir-generated.cpp: irgenerator $(ir_DEF_FILES)
+	$(V_IRGEN)$(SHELL) $(YLWRAP) _ ir-generated.cpp $@ ir-generated.h $(dir $@)ir-generated.h gen-tree-macro.h $(dir $@)gen-tree-macro.h -- ./irgenerator -i ir-generated.cpp -t gen-tree-macro.h -o ir-generated.h $(abspath $(ir_DEF_FILES))
```

Comment:
Whoops, that's an oversight. I'll upload a new version of this commit.

Actually, after thinking about it more this morning I think it'd be better to avoid `ylwrap` and just use a stamp file, like I'm doing in the unified compilation code in a later commit. That'll avoid the new to patch the IR generator, too.